flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosResourceManager.java

```diff
@@ -345,15 +382,20 @@ private void recoverWorkers() throws Exception {
 		CompletableFuture<Boolean> stopReconciliationCoordinatorFuture = stopActor(reconciliationCoordinator, stopTimeout);
 		reconciliationCoordinator = null;
```

Comment:
I think we need to check whether the supporting actors are not null because we initialize them only after the `clearStateFuture` has been completed. If this takes a bit and someone revokes our leadership in the meantime, `clearState` will be called before the support actors have been created.

Reply:
You are right. Fixed.